### PR TITLE
Update license-related files references

### PR DESCRIPTION
**Pull request by emmanuel-ferdman**
# PR Summary
Small PR - fixes the `LicenseScanTests.cs` and `LicenseExclusions.txt` references. It also fixes a few typos in `Documentation/UnifiedBuild/VMR-Permissible-Sources.md` on the way.

### CODE REVIEW

**Comment by missymessa** [comment]
@dotnet/source-build ^

File: Documentation/UnifiedBuild/VMR-Permissible-Sources.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ When a non-OSS license is detected, the offending code and binaries must be cloa
...
+Licenses are detected by the [license scan test](https://github.com/dotnet/dotnet/blob/main/test/Microsoft.DotNet.SourceBuild.Tests/LicenseScanTests.cs). This test is run as part of the [source-build license scan pipeline](https://dev.azure.com/dnceng/internal/_build?definitionId=1301&_a=summary) (internal Microsoft link). The test detects any license in the VMR that is not part of an exclusion listed in [`LicenseExclusions.txt`](https://github.com/dotnet/dotnet/blob/main/test/Microsoft.DotNet.SourceBuild.Tests/assets/LicenseScanTests/LicenseExclusions.txt).
 
 Common cases for adding a license to [`LicenseExclusions.txt`](https://github.com/dotnet/dotnet/blob/main/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets/LicenseExclusions.txt) include false positives, licenses related to test data, or needing to get a clean scan result with a relevant backport issue to remove the offending license later.
```
Comment (ellahathaway) [inline Documentation/UnifiedBuild/VMR-Permissible-Sources.md:94]:
```suggestion
Common cases for adding a license to [`LicenseExclusions.txt`](https://github.com/dotnet/dotnet/blob/main/test/Microsoft.DotNet.SourceBuild.Tests/assets/LicenseScanTests/LicenseExclusions.txt) include false positives, licenses related to test data, or needing to get a clean scan result with a relevant backport issue to remove the offending license later.
```

Same comment on line 96, too.

**Review by ellahathaway — COMMENTED** [review]
Thanks for addressing this :)

File: Documentation/UnifiedBuild/VMR-Permissible-Sources.md [hunk 2]
```diff
@@ -89,8 +89,8 @@ When a non-OSS license is detected, the offending code and binaries must be cloa
...
 
-Common cases for adding a license to [`LicenseExclusions.txt`](https://github.com/dotnet/dotnet/blob/main/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets/LicenseExclusions.txt) include false positives, licenses related to test data, or needing to get a clean scan result with a relevant backport issue to remove the offending license later.
+Common cases for adding a license to [`LicenseExclusions.txt`](https://github.com/dotnet/dotnet/blob/main/test/Microsoft.DotNet.SourceBuild.Tests/assets/LicenseScanTests/LicenseExclusions.txt) include false positives, licenses related to test data, or needing to get a clean scan result with a relevant backport issue to remove the offending license later.
 
 To have a license be added to the list of exclusions in [`LicenseExclusions.txt`](https://github.com/dotnet/dotnet/blob/main/test/Microsoft.DotNet.SourceBuild.SmokeTests/assets/LicenseExclusions.txt), open a PR, include a link to the relevant issue above the exclusion, and tag [@dotnet/source-build](https://github.com/orgs/dotnet/teams/source-build) as a reviewer.
```
Comment (ellahathaway) [inline Documentation/UnifiedBuild/VMR-Permissible-Sources.md:96]:
```Suggestion
To have a license be added to the list of exclusions in [`LicenseExclusions.txt`](https://github.com/dotnet/dotnet/blob/main/test/Microsoft.DotNet.SourceBuild.Tests/assets/LicenseScanTests/LicenseExclusions.txt), open a PR, include a link to the relevant issue above the exclusion, and tag [@dotnet/source-build](https://github.com/orgs/dotnet/teams/source-build) as a reviewer.
```

**Review by ellahathaway — APPROVED** [review]

**Review by mthalman — APPROVED** [review]

**Pull request merged**